build user model

baraah-berra opened the pull request.

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,97 @@
+  password: string;
+  role: 'admin' | 'cook' | 'customer';
+  isActive: boolean;
+  address: string;
```
address will be of type Address.

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,97 @@
+  address: string;
+}
+
+const addressSchema: Schema<Address> = new Schema({
+  city: {
+    type: String,
+    required: [true, 'City is required']
+  },
+  district: {
+    type: String,
+    required: [true, 'District is required']
+  },
+  neighborhood: {
+    type: String,
+    required: [true, 'Neighborhood is required']
+  },
+  streetAddress: {
+    type: String,
+    required: [true, 'Street address is required']
+  }
+});
```
no need to create a new schema for the address since the address will be just a nested object in the user model.

AsliSema replied:
I agree with you. No need to create new schema for address.

Khaled6120 reviewed (changes requested):
I just updated the way of implementing the address as we discussed earlier, u can refer to the DB design on Figma for better understanding.

**the updated code will look like this:**


![image](https://github.com/AsliSema/byte-and-bite/assets/86200305/680954bd-bcbe-47a4-83ce-f8c28d1e10cd)

![image](https://github.com/AsliSema/byte-and-bite/assets/86200305/f1dc2265-4823-44c0-8fdf-52d32e2a53e4)
![image](https://github.com/AsliSema/byte-and-bite/assets/86200305/8d55f325-f794-4ab7-aefb-d7ba744f9c3c)

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+import mongoose, { Document, Schema } from 'mongoose';
+
+// Define the user schema
+interface UserSchema extends Document {
```
Rename userSchema `IUser`
and export it
`export interface IUser extends Document {`

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+  phone: string;
+  profileImage: string;
+  password: string;
+  role: 'admin' | 'cook' | 'customer';
```
`role?: 'admin' | 'cook' | 'customer';`

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+  profileImage: string;
+  password: string;
+  role: 'admin' | 'cook' | 'customer';
+  isActive: boolean;
```
` isActive?: boolean;`

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+}
+
+
+const userSchema: Schema<UserSchema> = new Schema({
```
`const userSchema= new Schema<IUser>({`

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+      required: [true, 'Street address is required']
+    }
+  }
+});
```
add `{timestamp:true}`

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+    type: String,
+    required: [true, 'Email is required'],
+    unique: true,
+    validate: {
+        validator: (value: string) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value),
+        message: 'Invalid email format'
+    }
```
no need to implement complicated validator here in the schema, we will add a validation later for that later on.
schema should be simple as much as possible.
so i suggest deleting validate.

Khaled6120 commented on src/models/user.ts:
```diff
@@ -0,0 +1,92 @@
+  phone: {
+    type: String,
+    required: [true, 'Phone number is required'],
+    validate: {
+      validator: (value: string) => /^\d{10}$/.test(value),
+      message: 'Invalid phone number format'
+    }
```
no need to implement complicated validator here in the schema, we will add a validation later for that later on.
schema should be simple as much as possible.
so i suggest deleting validate.

Khaled6120 commented on package.json:
Please dont push package.json file in ur next push

Khaled6120 reviewed (changes requested):
Thanks Berra so much,
I have just added some comments and I will request changes, please do them they are minor changes.
Also just a reminder, dont push package.json file again, we have already installed all the packages that we need. for the next push, just push the user.ts file.

baraah-berra commented:
> Thanks Berra so much, I have just added some comments and I will request changes, please do them they are minor changes. Also just a reminder, dont push package.json file again, we have already installed all the packages that we need. for the next push, just push the user.ts file.

Thank you khaled, I reviewed and updated them and pushed user.ts code. I hope it is ready to merge now

Khaled6120 commented:
Berra I requested other changes, maybe u didnt notice them, plz check my review again and make the required changes.
Thanks for ur understanding.
